### ISSUE TITLE
Locally failing tests on modsec3-nginx

### ISSUE THREAD

**Issue by theseion** [coreruleset/coreruleset#2216]
Running tests for 941120 against modsec3-nginx fails for me locally (they don't fail against modec2-apache). The tests expect to see log entries that don't appear to be found, however, looking at the logs the entries are clearly there. Maybe the test is looking at the wrong logs?

* CRS version (e.g., v3.4/dev):
* Operating System and version: macOS 14

**Comment by airween** [comment]
[941120](https://github.com/coreruleset/coreruleset/blob/v3.4/dev/tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941120.yaml) has 4 tests - all of them are failed?

Other test cases are fine?

With which test framework do you use? [go-ftw](https://github.com/fzipi/go-ftw) or [ftw](https://github.com/coreruleset/ftw)?

**Comment by theseion** [comment]
The first two fail because they expect the ID to be in the log. The other two succeed because they expect no ID in the log.

The same applies to other tests. Whenever the ID is expected to be in the log, the test fails. This is using ftw.

It's probably something related to my setup but I wanted to check if I'm the only one who has this problem first.

BTW, should I even invest time in this? Or is go-ftw ready for prime time? Are there instructions on how to run tests with go-ftw locally?

**Comment by airween** [comment]
> The same applies to other tests. Whenever the ID is expected to be in the log, the test fails. This is using ftw.

If the entry is there (in the log), but `ftw` does not find that, it could be some timezone issue on your system. Check that Nginx produces the logs with correct hour, what your system uses.

**Comment by fzipi** [comment]
@theseion I've been using go-ftw since a long time, and it works all the time for me. But of course I'm cheating because I'm the author. Sometimes, depending on the type of execution you are doing (container vs. base host), you need to check the timezone.

Go-ftw will fail because the timezone can't be detected properly and you will see that all log related tests fail (like this case). The typical usage scenario that both test tool do is to save _t0_ before the test and _t1_ after the test, and then check logs between them. I've created https://github.com/fzipi/go-ftw/issues/40 some time ago to give users better output when this happens.

Take a look on how it's being [used in the pipeline](https://github.com/coreruleset/coreruleset/blob/v3.4/dev/.github/workflows/go-ftw.yml), you can easily do the same. And if you have any doubts, or comments, you are more than welcome to file an issue or ping me in slack ;)

**Comment by theseion** [comment]
Thanks guys. You seem to be on the right track. The timezone wasn't correct (why doesn't this happen in the modesec2-apache container?). However, `TZ` isn't set in my shell (and usually isn't), so it can't be picked up that way. Setting `TZ` manually helped to fix the timestamps in the logs. However, the tests still don't work.

I noticed that the container printed a warning (which I had missed...) about `TZ` not being set. The interesting thing is that it is being printed twice, but after I set `TZ` it is *still being printed once`.

An alternative to setting `TZ` could be to mount `/etc/localtime:/etc/localtime:ro`, but that obviously has some limitations.

I haven't spent much time on this. I'll come back to it later.
